src/components/map-page-components/map-view/MapView.tsx

```diff
@@ -1,7 +1,203 @@
+  map = new google.maps.Map(document.getElementById('map') as HTMLElement, {
+    center: BOSTON,
+    zoom: 12,
+    fullscreenControl: false,
+    mapTypeControl: false,
+    // mapId: 76c08a2450c223d9,
+    restriction: {
+      latLngBounds: BOSTON_BOUNDS,
+      strictBounds: false,
+    },
+  });
+
+  const input = document.getElementById('pac-input') as HTMLInputElement;
+  const autocomplete = new google.maps.places.Autocomplete(input);
+  autocomplete.addListener('place_changed', () => {
+    const place = autocomplete.getPlace();
+
+    if (!place.geometry) {
+      window.alert("No details available for input: '" + place.name + "'");
+      return;
+    }
+
+    // If the place has a geometry, then present it on a map.
+    if (place.geometry.viewport) {
+      map.fitBounds(place.geometry.viewport);
+    } else {
+      map.setCenter(place.geometry.location);
+      map.setZoom(16);
+    }
+  });
+
+  const privateStreetsLayer = new google.maps.Data({ map });
+
+  privateStreetsLayer.loadGeoJson(
+    'https://raw.githubusercontent.com/florisdobber/SFTT-map-test/master/private_streets.json',
+  );
+
+  function setPrivateStreetsStyle(v: boolean) {
+    privateStreetsLayer.setStyle({
+      strokeColor: 'red',
+      strokeWeight: 2,
+      visible: v,
+    });
+  }
+
+  setPrivateStreetsStyle(false);
+
+  const blocksLayer = new google.maps.Data({ map });
+  blocksLayer.loadGeoJson(
+    'https://raw.githubusercontent.com/florisdobber/SFTT-map-test/master/blocks.json',
+  );
+
+  function setBlocksStyle(v: boolean) {
+    blocksLayer.setStyle((feature) => {
+      let color = 'green';
+
+      // Use this for coloring reserved/completed blocks a different color
+      if (feature.getProperty('ID') % 10 === 0) {
+        color = 'yellow';
+      }
+
+      if (feature.getProperty('ID') % 10 === 1) {
+        color = 'red';
+      }
+
+      // Use this for selecting blocks
+      if (feature.getProperty('PRECINCT') === 'YES') {
+        color = 'black';
+      }
+
+      // Set styling here
+      return {
+        fillColor: color,
+        strokeColor: 'black',
+        strokeWeight: 1,
+        visible: v,
+      };
+    });
+  }
+
+  setBlocksStyle(false);
+
+  const neighborhoodsLayer = new google.maps.Data({ map });
+  neighborhoodsLayer.loadGeoJson(
+    'https://raw.githubusercontent.com/florisdobber/SFTT-map-test/master/neighborhoods_edited.geojson',
+  );
+
+  function setNeighborhoodsStyle(v: boolean) {
+    neighborhoodsLayer.setStyle((feature) => {
+      return {
+        fillColor: 'green',
+        fillOpacity: (feature.getProperty('Neighborhood_ID') / 100) * 2 + 0.1, // replace this with completion percentage
+        strokeWeight: 1,
+        strokeColor: 'white',
+        visible: v,
+      };
+    });
+  }
+
+  setNeighborhoodsStyle(true);
+
+  // Check for clicks on neighborhoods and zoom to when clicked on a neighborhood
+  neighborhoodsLayer.addListener('click', (event) => {
+    map.setZoom(14);
```

Comment:
Does the google maps API offer a constant for zoom levels?

Reply:
I don't think so, everyone does it like this.